codegen/mustache/java/libraries/okhttp-gson/ApiClient.mustache

```diff
@@ -0,0 +1,1871 @@
+
+        return contentTypes[0];
+    }
+
+    /**
+     * Escape the given string to be used as URL query value.
+     *
+     * @param str String to be escaped
+     * @return Escaped string
+     */
+    public String escapeString(String str) {
+        try {
+            return URLEncoder.encode(str, "utf8").replaceAll("\\+", "%20");
+        } catch (UnsupportedEncodingException e) {
+            return str;
+        }
+    }
+
+    /**
+     * Deserialize response body to Java object, according to the return type and
+     * the Content-Type response header.
+     *
+     * @param <T> Type
+     * @param response HTTP response
+     * @param returnType The type of the Java object
+     * @return The deserialized Java object
+     * @throws {{invokerPackage}}.ApiException If fail to deserialize response body, i.e. cannot read response body
+     *   or the Content-Type of the response is not supported.
+     */
+    @SuppressWarnings("unchecked")
+    public <T> T deserialize(Response response, Type returnType) throws ApiException {
+        if (response == null || returnType == null) {
+            return null;
+        }
+
+        if ("byte[]".equals(returnType.toString())) {
+            // Handle binary response (byte array).
+            try {
+                return (T) response.body().bytes();
+            } catch (IOException e) {
+                throw new ApiException(e);
+            }
+        } else if (returnType.equals(File.class)) {
+            // Handle file downloading.
+            return (T) downloadFileFromResponse(response);
+        }
+
+        String respBody;
+        try {
+            if (response.body() != null)
+                respBody = response.body().string();
+            else
+                respBody = null;
+        } catch (IOException e) {
+            throw new ApiException(e);
+        }
+
+        if (respBody == null || "".equals(respBody)) {
+            return null;
+        }
+
+        String contentType = response.headers().get("Content-Type");
+        if (contentType == null) {
+            // ensuring a default content type
+            contentType = "application/json";
+        }
+        if (isJsonMime(contentType)) {
+            // retrieve location from headers
+            String location = response.headers().get("Location");
+            if (location != null) {
+                final String regex = "\\/([^\\/]+)\\/?$";
+                final Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+                final Matcher matcher = pattern.matcher(location);
+                try {
+                    matcher.find();
+                    JsonObject jsonObject = JsonParser.parseString(respBody).getAsJsonObject();
+                    jsonObject.addProperty("docId", matcher.group(1));
+                    return JSON.deserialize(jsonObject.toString(), returnType);
+                } catch (Exception e) {
+                    System.out.println("respBody is:" + respBody);
```

Comment:
This hidden error message could be confusing for customers without context of why it was done. I'd suggest rewording to:

```
System.out.println("Expected JSON response body was not received. Actual response body was `" + respBody + "`");
```